localstack-core/localstack/services/apigateway/next_gen/execute_api/handlers/integration_response.py

```diff
@@ -61,22 +73,57 @@ def __call__(
             response_parameters=response_parameters,
         )
 
-        # we can also apply the Response Templates, they will depend on status code + accept header?
-
-        # then responseOverride
+        # We then fetch a response templates and apply the template mapping
+        response_template = self.get_response_template(
+            integration_response=integration_response, request=context.invocation_request
+        )
+        body, response_override = self.render_request_template_mapping(
+            context=context, template=response_template, response=response
+        )
 
         # here we update the `Response`. We basically need to remove all headers and replace them with the mapping, then
         # override them if there are overrides.
-        # for the body, it will maybe depend on configuration?
-        for key in list(response.headers.keys(lower=True)):
-            if key not in ("connection", "content-type"):
-                del response.headers[key]
+        # The status code is pretty straight forward. By default, it would be set by the integration response,
+        # unless there was an override
+        response.status_code = int(integration_response["statusCode"])
+        if response_status_override := response_override["status"]:
+            # maybe make a better error message format, same for the overrides for request too
+            LOG.debug("Overriding response status code: '%s'", response_status_override)
+            response.status_code = response_status_override
+
+        # Create a new headers object that we can manipulate before overriding the original response headers
+        headers = Headers(response_data_mapping.get("header"))
+        if header_override := response_override["header"]:
+            LOG.debug("Response header overrides: %s", header_override)
+            headers.update(header_override)
+
+        # When trying to override certain headers, they will instead be remapped
+        # There may be other headers, but these have been confirmed on aws
+        remapped_headers = (
+            "connection",
+            "content-length",
+            "date",
+            "x-amzn-requestid",
+            "content-type",
+        )
+        for header in remapped_headers:
+            if value := headers.get(header):
+                headers[f"x-amzn-remapped-{header}"] = value
+                headers.remove(header)
 
-        # there must be some default headers set, snapshot those?
-        if mapped_headers := response_data_mapping.get("header"):
-            response.headers.update(mapped_headers)
+        # Those headers are passed through from the response headers, there might be more?
+        passthrough_headers = ("connection", "content-type", "content-length")
```

Comment:
maybe `Content-Type` might not come from the response but is set here, depending on the template? We'll need to try with an integration that returns XML or HTML but the response template is JSON or there is no response template 🤔 or maybe you've done it already 😅 

> API Gateway includes a Content-Type header for all integration responses. By default, the content type is application/json.